frontend/src/muralForm/MuralForm.tsx

```diff
@@ -15,58 +20,113 @@ const useStyles = makeStyles((theme: Theme) =>
       margin: theme.spacing(1),
       width: "195px",
     },
+    bottomButton: {
+      fontWeight: 500,
+    },
+    bottomButtonContainer: {
+      display: "flex",
+      justifyContent: "space-between",
+    },
   })
 );
 
 interface IMuralFormProps {}
 
 function MuralForm(props: IMuralFormProps) {
   const styles = useStyles();
-  const currYear = new Date().getFullYear();
+  const [name, setName] = useState<string>("");
+  const [year, setYear] = useState<number>(new Date().getFullYear());
   const [assistants, setAssistants] = useState<string[]>([]);
   const [socialMedia, setSocialMedia] = useState<string[]>([]);
-  //todo: delete these logs once we actually use the state
-  console.log(assistants)
-  console.log(socialMedia)
+  const [artist, setArtist] = useState<number | null>(null);
+  const [borough, setBorough] = useState<number | null>(null);
+  const [addressCoords, setAddressCoords] = useState<number[]>([]);
+  const [description, setDescription] = useState<string>("");
+  const [neighbourhood, setNeighbourhood] = useState<string>("");
+  const [address, setAddress] = useState<string>("");
 
-  function updateAssistants(newAssistants: string[]) {
-    setAssistants(newAssistants);
+  function submitForm() {
+    if (name === "") {
+      alert("Please enter a name.");
+      return;
+    }
+    if (artist == null) {
+      alert("Please select a valid artist.");
+      return;
+    }
+    if (borough == null) {
+      alert("Please select a valid borough.");
+      return;
+    }
+    if (addressCoords.length !== 2 || address === "" || neighbourhood === "") {
+      alert("Please check validity of address.");
+      return;
+    }
+    axios
+      .post(CREATE_MURAL_API, {
+        name: name,
+        boroughId: borough,
+        artistId: artist,
+        year: year,
+        city: "Montreal",
+        longitude: addressCoords[0],
+        latitude: addressCoords[1],
+        partners: assistants,
+        description: description,
+        socialMedia: socialMedia,
+        address: address,
+        neighbourhood: neighbourhood,
+      })
+      .then(
+        (response) => {
+          //TODO notify success and clear form.
+          console.log(response);
+        },
+        (error) => {
+          console.log(error);
+        }
+      );
   }
 
-  function updateSocialMedia(newSocialMedia: string[]) {
-    setSocialMedia(newSocialMedia);
+  function handleAddressUpdate(
+    coords: number[],
+    address: string,
+    neighbourhood: string
+  ) {
+    setAddressCoords(coords);
+    setAddress(address);
+    setNeighbourhood(neighbourhood);
   }
 
   return (
     <form noValidate autoComplete="off">
       <div className={styles.flexContainer}>
         <TextField
           className={styles.element}
-          required
+          required={true}
           id="name"
           label="Name"
           variant="filled"
           size="small"
+          onChange={(e: any) => setName(e.target.value)}
```

Comment:
are these events of type `SyntheticEvent`?

Reply:
I think ChangeEvent<HTMLInputElement>